runtime/src/bank.rs

```diff
@@ -3178,6 +3165,20 @@ impl Bank {
         consumed_budget.saturating_sub(budget_recovery_delta)
     }
 
+    // This is called from snapshot restore and for each epoch boundary
+    // The entire code path herein must be idempotent
+    pub fn refresh_programs_and_inflation(&mut self) {
```

Comment:
NIT:  I have a feeling this name won't age well.  Maybe something like `apply_feature_activations()` ?